src/ui/components/Widgets/Chart/Chart.tsx

```diff
@@ -216,6 +219,26 @@ export const Chart = (props: ChartNoWidgetProps) => {
             ],
         );
 
+    // handle reload chart event for iframe preview
+    React.useEffect(() => {
+        const isEmbedded = isEmbeddedMode();
+        const handleMessageEvent = (event: MessageEvent) => {
+            if (event.data.type === CHART_RELOAD_EVENT) {
+                reloadChart();
+            }
+        };
+
+        if (isEmbedded) {
+            window.addEventListener('message', handleMessageEvent);
```

Comment:
Maybe move return here and do not check on every reloadChart ref update it for no reason?